### DISCUSSION TITLE
Staged rollout of a new extension version

### DISCUSSION THREAD

**Discussion by hyangah** [microsoft/vscode-discussions#365] [Extension Development QnA] [answered]
According to the [doc](https://code.visualstudio.com/docs/supporting/FAQ#_opt-out-of-extension-updates) vs code will auto-update extensions
as soon as the new versions become available if the user does not opt out of the auto-update feature.
As an extension publisher, I am often worried about the scale of damage a bad release can cause.
We can emphasize thorough testing before release and run a longer pre-release period, but
the idea of affecting such a large group of users at once is scary.

Is there a way for extension publishers to control the rollout speed?
e.g. push to 10% for day1, 40% for day2,  50% for day7, ...

We may want to roll out minor version update or critical security patch more aggressively,
while we want to go slow for major version.
And it would be nice if we can pause the rollout when a serious issue is detected.

**EDIT: Based on the answer, I finally opened a feature request in https://github.com/microsoft/vsmarketplace/issues/550.
Please upvote if you are interested.**

**Comment by alefragnani** [comment, marked as answer]
Hi @hyangah ,

No, you can't _control_ the rollout. Once you publish the extension, and it passes the verification process of the marketplace, the new release will be available for every user.

The only official solution, for now, is [Pre-release extensions](https://code.visualstudio.com/api/working-with-extensions/publishing-extension#prerelease-extensions). Or, you could publish a second _Insiders_ extension, like GitLens does (https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens-insiders).

BTW, there is an open issue in the VS Marketplace repo, requesting to _unpublish a specific version_ of an extension https://github.com/microsoft/vsmarketplace/issues/235. I suggest you to upvote there.

Hope this helps
